src/telegram/_utils/argumentparsing.py

```diff
@@ -50,6 +51,22 @@ def parse_sequence_arg(arg: Optional[Sequence[T]]) -> tuple[T, ...]:
     return tuple(arg) if arg else ()
 
 
+def parse_period_arg(arg: Optional[Union[int, float, dtm.timedelta]]) -> Optional[dtm.timedelta]:
+    """Parses an optional time period in seconds into a timedelta
+
+    Args:
+        arg (:obj:`int` | :class:`datetime.timedelta`, optional): The time period to parse.
+
+    Returns:
+        :obj:`timedelta`: The time period converted to a timedelta object or :obj:`None`.
+    """
+    if arg is None:
+        return None
+    if isinstance(arg, (int, float)):
+        return dtm.timedelta(seconds=arg)
+    return arg
```

Comment:
the type checker is probably complaining when using this function because there's a chance we could don't return None or a timedelta, which contradicts the expected return type of the function. Maybe `arg` should accept a generic too? The resulting Returns docstring needs to be changed too. cc @Bibo-Joshi